### ISSUE TITLE
Builder scripts using global `npm` dependencies instead of locals

### ISSUE THREAD

**Issue by dandelany** [FormidableLabs/builder#47]
Opening another ticket here per @boygirl 's request... See FormidableLabs/victory#97 for more details & examples.

While I was working on Victory, I discovered that `builder` scripts which call node binaries like `babel` end up using the globally installed versions of those binaries, rather than the local versions in `node_modules`. This is contrary to the behavior of regular old `npm` scripts which use the local versions. This can cause major problems and build inconsistencies if the user's global module version does not match the local version.

**Comment by ryan-roemer** [comment]
So this behavior likely comes from: https://github.com/FormidableLabs/builder/blob/master/lib/environment.js#L68-L72

```js
Environment.prototype.updateNodePath = function (archetypeNodePaths) {
  return (this.env.NODE_PATH || "")
    .split(DELIM)
    .filter(function (x) { return x; }) // Remove empty strings
    .concat([CWD_NODE_PATH].concat(archetypeNodePaths || []))
    .join(DELIM);
};
```

PROPOSAL: Resolve in this order:

* Current Working directory
* Archetype directories
* _then_ pre-existing `NODE_PATH`

How does that sound?

/cc @dandelany @boygirl @exogen @chaseadamsio

**Comment by dandelany** [comment]
That sounds pretty good to me! Of course, that order would still allow the user's local project to override the Archetype's dependency version... But maybe that is preferred, so that user project's can consciously extend the archetype... I'm still too much of a `builder` n00b to be sure :)

**Comment by ryan-roemer** [comment]
Released as `builder@2.1.4`
